sonic-xcvrd/xcvrd/xcvrd.py

```diff
@@ -507,62 +538,44 @@ def post_port_dom_info_to_db(logical_port_name, port_mapping, table, stop_event=
-                         ('tx7power', dom_info_dict['tx7power']),
-                         ('tx8power', dom_info_dict['tx8power'])
-                         ])
-                else:
-                    fvs = swsscommon.FieldValuePairs(
-                        [('temperature', dom_info_dict['temperature']),
-                         ('voltage', dom_info_dict['voltage']),
-                         ('rx1power', dom_info_dict['rx1power']),
-                         ('rx2power', dom_info_dict['rx2power']),
-                         ('rx3power', dom_info_dict['rx3power']),
-                         ('rx4power', dom_info_dict['rx4power']),
-                         ('tx1bias', dom_info_dict['tx1bias']),
-                         ('tx2bias', dom_info_dict['tx2bias']),
-                         ('tx3bias', dom_info_dict['tx3bias']),
-                         ('tx4bias', dom_info_dict['tx4bias']),
-                         ('tx1power', dom_info_dict['tx1power']),
-                         ('tx2power', dom_info_dict['tx2power']),
-                         ('tx3power', dom_info_dict['tx3power']),
-                         ('tx4power', dom_info_dict['tx4power'])
-                         ])
-
-                table.set(port_name, fvs)
-
+                fvs = swsscommon.FieldValuePairs([(k, v) for k, v in dom_info_dict.items()])
+                table.set(physical_port_name, fvs)
             else:
                 return SFP_EEPROM_NOT_READY
 
         except NotImplementedError:
             helper_logger.log_error("This functionality is currently not implemented for this platform")
             sys.exit(NOT_IMPLEMENTED_ERROR)
 
+# Update port pm info in db
+
+
+def post_port_pm_info_to_db(logical_port_name, port_mapping, table, stop_event=threading.Event(), pm_info_cache=None):
+    for physical_port, physical_port_name in get_physical_port_name_dict(logical_port_name, port_mapping).items():
+        if stop_event.is_set():
+            break
+
+        if not _wrapper_get_presence(physical_port):
+            continue
+
+        if pm_info_cache is not None and physical_port in pm_info_cache:
+            # If cache is enabled and pm info is in cache, just read from cache, no need read from EEPROM
+            pm_info_dict = pm_info_cache[physical_port]
+        else:
+            pm_info_dict = _wrapper_get_transceiver_pm(physical_port)
+            if pm_info_cache is not None:
+                # If cache is enabled, put dom information to cache
+                pm_info_cache[physical_port] = pm_info_dict
+        if pm_info_dict is not None:
+            # Skip if empty (i.e. get_transceiver_pm API is not applicable for this xcvr)
+            if not pm_info_dict:
+                continue
+            beautify_pm_info_dict(pm_info_dict, physical_port)
+            fvs = swsscommon.FieldValuePairs([(k, v) for k, v in pm_info_dict.items()])
+            table.set(physical_port_name, fvs)
```

Comment:
can we keep the key as logical port just like done for DOM and other table?

Reply:
This is because the CLI and streaming telemetry will need logical port as key in future when we do breakout of physical port to many logical ports.

Reply:
@keboliu if you check the usage of port_name in previous change:-

table.set(port_name, fvs) 

port_name is nothing but physical port name at line 496 in previous code :- 

port_name = get_physical_port_name(logical_port_name, ganged_member_num, ganged_port)

so, i don't see this PR causing any issue. We need to revisit once we support breakout in Xcvrd.